src/Upgrader/Commands/Upgrade/UpgradeCommand.php

```diff
@@ -82,6 +82,11 @@ public function execute(ContextInterface $context): ContextInterface
         $message = new Message((string)$stepsExecutionDto->getOutputMessage());
 
         $context->addMessage(static::class, $message);
+        $context->setExitCode(ContextInterface::SUCCESS_EXIT_CODE);
+
+        if (!$stepsExecutionDto->getIsSuccessful()) {
+            $context->setExitCode(ContextInterface::FAILURE_EXIT_CODE);
```

Comment:
Those constants are not named as per const guidelines. But I guess nothing that can now be changed anymore.